hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/lz4/Lz4Decompressor.java

```diff
@@ -67,6 +55,15 @@
   public Lz4Decompressor(int directBufferSize) {
     this.directBufferSize = directBufferSize;
 
+    try {
+      LZ4Factory lz4Factory = LZ4Factory.fastestInstance();
+      lz4Decompressor = lz4Factory.safeDecompressor();
+    } catch (Throwable t) {
```

Comment:
Shame the constructor can't throw exceptions direct.

Catch only the exception's known to be raised, and don't convert RTEs or, especially, Errors.

Reply:
Hmm, I tried to explicitly catch `ClassNotFoundException`, but the java compiler complains...

```
 exception java.lang.ClassNotFoundException is never thrown in body of corresponding try statement
```

Reply:
LZ4Factory seems to throw java.lang.AssertionError after catching unrecoverable exception.

Reply:
ok, so what's best here?
1. Catch AssertionError and wrap
2. Catch throwable, but with something ahead of it which will catch and rethrow Error without wrapping. Because we shouldn't really be wrapping those high-priority problems that aren't generally things to ignore

Reply:
Let me just catch `AssertionError` and wrap it.